controllers/controlplane_controller.go

```diff
@@ -32,9 +40,44 @@ type ControlPlaneReconciler struct {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	// we need to get the old objects for cluster-wide resource updates to address
+	// cases in which the ownerReference is changed or deleted. If we use the newObject,
+	// we cannot infer the object owner, as there is no reference to it, and we'd lose
+	// the event that should trigger the creation of a new resource
```

Comment:
:+1: I appreciate the comment here. I do wonder though if I may suggest a rewording that I think might be a bit more straightforward?

```suggestion
	// for owned objects we need to check if updates to the objects resulted in the
	// removal of an OwnerReference to the parent object, and if so we need to
	// enqueue the parent object so that reconcilation can create a replacement.
```

Just a thought, let me know your thoughts? Not a blocker, resolve at your own discretion.